tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc

```diff
@@ -1385,6 +1424,23 @@ Status Converter::BuildCudaEngine(
   TrtUniquePtrType<nvinfer1::IBuilderConfig> builder_config(
       trt_builder_->createBuilderConfig());
   builder_config->setMaxWorkspaceSize(max_workspace_size_bytes);
+
+#if IS_TRT_VERSION_GE(7, 1, 3, 0)
+   // Only activate TRT deterministic mode if there is a requested tactic ID
+   static bool trt_debug_deterministic = [] {
+     int64 trt_tactic_id = -1;
+     TF_CHECK_OK(tensorflow::ReadInt64FromEnvVar(
+       "TF_DEBUG_TRT_DETERMINISTIC_TACTIC_ID",
+       /*default_val=*/-1, &trt_tactic_id));
+     return (trt_tactic_id >= 0);
+   }();
+
+  if (trt_debug_deterministic) {
+    DeterministicAlgorithmSelector algo_selector;
+    builder_config->setAlgorithmSelector(&algo_selector);
+  }
```

Comment:
Let's rename the class to StaticAlgorithmSelector, because it only select the algorithm with a fixed ID.
We can change the constructor of the StaticAlgorithmSelector to take an int64, to avoid reading the environment variable again. In this case, the code block can be rewritten as following with the lambda called in place:

int64 algorithm_id = [] {
    int64 algorithm_id;
    TF_CHECK_OK(tensorflow::ReadInt64FromEnvVar("TF_TRT_ALGORITHM_ID", /*default_val=*/-1, &algorithm_id));
   return algorithm_id;
}();

if (algorithm_id >=0) {
    StaticAlgorithmSelector algo_selector(algorithm_id);
    builder_config->setAlgorithmSelector(&algorithm_selector);
}